### DISCUSSION TITLE
Using old value of AD variable in ADMaterial

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27014] [Q&A General] [answered]
Hi all,

I need to use the current and the old value of an AD variable (`void_fraction`) in ADMaterial, so I just define:

```
  _void_fraction(adCoupledValue("void_fraction")),
  _void_fraction_old(coupledValueOld("void_fraction")),
```

I would like to know if there is something wrong in ADMaterial since the `_void_fraction_old` is not AD type. I think adCoupledValueOld is  better but there is no such type. Could anyone kindly tell me? Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

There are no 'old' dofs in the current equation system so we do not need to track the dependency of the material properties with regards to old dofs. Hence why we dont use AD for old and older properties. AD is only for the derivatives with regards to current DOFs

Guillaume

**Reply by SomnusYu** [reply]
Because "void_fraction" is a variable, not a material property. I need to get the old value of this AD variable, so I use `coupledValueOld`.

**Reply by SomnusYu** [reply]
Should I define an AD material property (_void_property), and let `_void_property[_qp] = _void_fraction[_qp]`. Then I get the old value using `_void_property(getADMaterialPropertyOld<Real>(void_fraction))`?

**Reply by GiudGiud** [reply]
coupledValueOld is fine. We dont need AD for old. 

> Should I define an AD material property (_void_property), and let _void_property[_qp] = _void_fraction[_qp]. Then I get the old value using _void_property(getADMaterialPropertyOld<Real>(void_fraction))?

this is not going to do anything good. It will make it look AD but the derivatives will still be 0 because you're looking at the old material property now

**Reply by SomnusYu** [reply]
I got it, thanks!
